scheduler/scheduler.go

```diff
@@ -582,25 +710,23 @@ func (driver *MesosSchedulerDriver) doReliableRegistration(maxBackoff float64) {
 // return true if we should attempt another registration later
 func (driver *MesosSchedulerDriver) registerOnce() (proceed bool) {
 
-	proceed = true
 	var (
 		failover bool
 		pid      *upid.UPID
 	)
-
-	func() {
+	if func() bool {
 		driver.lock.RLock()
 		defer driver.lock.RUnlock()
 
 		//TODO(jdef) also, check that we know which master to send to; if none, return false
-		//TODO(jdef) also, if authentication is also asynchronous, check if we have credentials && !authenticated, if so, return false
-		if driver.stopped || driver.connected {
-			proceed = false
+		if driver.stopped || driver.connected || (driver.credential != nil && !driver.authenticated) {
```

Comment:
in test statement, will the (driver.credential != nil && !driver.authenticated) ever be true ? Back in https://github.com/jdef/mesos-go/blob/zk2/scheduler/scheduler.go#L292 driver.authenticated = true when driver.credential = nil.